### PR TITLE
MGDOBR-1211: add processors api V2

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-1211](https://issues.redhat.com/browse/MGDOBR-1211)

Add processor v2 api

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Review by manstis — APPROVED** [review]
LGTM 👍

**Review by ChughShilpa — APPROVED** [review]

File: openapi/smartevents_mgmt_v2.yaml [hunk 1]
```diff
@@ -281,6 +284,289 @@ paths:
...
       security:
       - bearer: []
+  /api/smartevents_mgmt/v2/bridges/{bridgeId}/processors:
+    get:
+      tags:
+      - Processors
+      summary: Get the list of Processors of a Bridge instance
+      description: Get the list of Processors of a Bridge instance for the authenticated
+        user.
+      operationId: ProcessorsAPI_listProcessors
```
Comment (kelvah) [inline openapi/smartevents_mgmt_v2.yaml:294]: Can we keep the same verbs of the bridge API? like `ProcessorAPI_getProcessors`. 
They end up as names of the sdk client methods.

File: openapi/smartevents_mgmt_v2.yaml [hunk 2]
```diff
@@ -281,6 +284,289 @@ paths:
...
+              schema:
+                $ref: '#/components/schemas/ErrorsList'
+      security:
+      - bearer: []
+    post:
+      tags:
+      - Processors
+      summary: Create a Processor of a Bridge instance
+      description: Create a Processor of a Bridge instance for the authenticated user.
+      operationId: ProcessorsAPI_addProcessorToBridge
```
Comment (kelvah) [inline openapi/smartevents_mgmt_v2.yaml:365]: `operationId: ProcessorsAPI_createProcessor`, similarly to `createBridge`?

File: openapi/smartevents_mgmt_v2.yaml [hunk 3]
```diff
@@ -880,6 +1166,87 @@ components:
...
+          allOf:
+          - $ref: '#/components/schemas/ManagedResourceStatus'
+          - nullable: false
+        owner:
+          type: string
+          nullable: false
+        name:
+          type: string
+        flows:
+          type: object
```
Comment (kelvah) [inline openapi/smartevents_mgmt_v2.yaml:1249]: Should they be not nullable too?

**Review by kelvah — COMMENTED** [review]
@r00ta left some comments

**Review by kelvah — APPROVED** [review]
Thanks for the changes! :+1:

**Pull request merged**